datachannel.go

```diff
@@ -228,6 +230,34 @@ func (d *DataChannel) onOpen() {
 	}
 }
 
+// OnDial sets an event handler which is invoked when the
+// peer has been dialed, but before said peer has responsed
+func (d *DataChannel) OnDial(f func()) {
+	d.mu.Lock()
+	d.dialHandlerOnce = sync.Once{}
+	d.onDialHandler = f
+	d.mu.Unlock()
+
+	if d.ReadyState() == DataChannelStateOpen {
```

Comment:
Added a test; its a bit unrealistic, but ensures this handler will always be called. The gist is that if the `OnDial` handler is added after the dial event occurs, it would never be called without this check. `OnOpen` does something similar